### PR TITLE
added fixture to capture deadlock logs

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

In relation to #6117 and #6090 

1. I was unable to reproduce 
1. it is unclear to me how postgresql takes locks when doing a delete cascade 
 - In general, locks should be acquired consistently with delete cascade lock order, which is top down.  This means
   -  agentprocess -> agentinstance
   - environment -> agentprocess
   - environment -> agent
  2. the agent manager assumes agentprocess -> agentinstance -> agent
  3. I found one ordering bug in the agent manager, but I don't know if that was it
  4. I made it so the postgresql log part after the deadlock is logged, that is like
  
  ```15:37:49.068 WARNING 2023-06-15 15:29:43.830 CEST [75118] DETAIL:  Process 75118 waits for AccessExclusiveLock on relation 16571 of database 16385; blocked by process 75121.
15:37:49.068 WARNING 	Process 75121 waits for RowExclusiveLock on relation 16599 of database 16385; blocked by process 75118.
15:37:49.068 WARNING 	Process 75118: LOCK TABLE public.agentprocess
15:37:49.068 WARNING 	Process 75121: LOCK TABLE agent IN ROW EXCLUSIVE MODE
15:37:49.068 WARNING 2023-06-15 15:29:43.830 CEST [75118] HINT:  See server log for query details.
15:37:49.068 WARNING 2023-06-15 15:29:43.830 CEST [75118] STATEMENT:  LOCK TABLE public.agentproces

```

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/server/agentmanager.py [hunk 1]
```diff
@@ -411,10 +411,10 @@ async def _log_session_seen_to_db(
         now = datetime.now().astimezone()
         async with data.AgentProcess.get_connection() as connection:
             async with connection.transaction():
+                await data.AgentProcess.update_last_seen(session.id, now, connection)
```
Comment (arnaudsjs) [inline src/inmanta/server/agentmanager.py:414]: If I understand everything correctly. The locking order is: `environment->agent->agentprocess->agentinstance`. Do you agree? In that case we should move the last line to the top as well.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/server/agentmanager.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/server/agentmanager.py:414]: agentprocess->agentinstance -> agent 

is what is used in the rest of the agentmanager.

For the cascading delete, I don't know.

This at least makes it consistent with the rest of the file

Comment (arnaudsjs) [reply src/inmanta/server/agentmanager.py:414]: Hmm. In the `Environment.delete_cascade()` method, the `Agent` is delete first and only then the `AgentProcess`. I am not convinced the locking order is right now, but maybe it's better to move forward with little steps and to see what happens.

**Review by arnaudsjs — APPROVED** [review]

File: src/inmanta/server/agentmanager.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/server/agentmanager.py:414]: In the first part of the method: yes, but no transaction there. 

The second part: I don;t know.

Comment (arnaudsjs) [reply src/inmanta/server/agentmanager.py:414]: Indeed. I completely missed the fact that it doesn't run in a transaction.

**Pull request closed without merging**